File: 프로그래머스/0/120890. 가까운 수/가까운 수.js
function solution(array, n) {
    // let answer = Math.abs(n-array[0]);
    // for(let i=0; i<array.length; i++) {
    //     if(Math.abs(n-array[i])<answer) {
    //         answer = array[i];
    //     }
    // }
    // return answer;
    
    array.sort((a, b) => a-b);
    let a = 0;
    let b = 0;
    let answer = [];

    for (let i = 0; i < array.length; i++) {
        answer.push(Math.abs(n-array[i]));
        a = Math.min(...answer);
        b = answer.indexOf(a);
    }
    return array[b];
}

File: 프로그래머스/0/120890. 가까운 수/가까운 수.java
class Solution {
    public int solution(int[] array, int n) {
        int answer = 0;
        int min = 0;
        for(int num : array){
            if(answer == 0){
                answer = num;
                min = Math.abs(num - n);
            }else{
                if(Math.abs(num - n) < min){
                    answer = num;
                    min = Math.abs(num - n);
                }else if( Math.abs(num - n) == min && num < answer){
                    answer = num;
                }
            }
        }
        return answer;
    }
}